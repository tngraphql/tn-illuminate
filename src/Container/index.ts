/**
 * (c) Phan Trung NguyÃªn <nguyenpl117@gmail.com>
 * User: nguyenpl117
 * Date: 3/11/2020
 * Time: 12:32 PM
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 */

export { Container } from './Container';
export { Injector } from './Injector';
export { InvalidArgumentException } from './InvalidArgumentException';
export { LogicException } from './LogicException';
export { BindingResolutionException } from './BindingResolutionException';