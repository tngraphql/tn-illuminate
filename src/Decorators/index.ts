/**
 * (c) Phan Trung NguyÃªn <nguyenpl117@gmail.com>
 * User: nguyenpl117
 * Date: 3/21/2020
 * Time: 4:51 PM
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 */

export * from './Inject';
export * from './Rules';
export * from './Service';
export * from './UseMiddleware';
export * from './ValidateArgs';