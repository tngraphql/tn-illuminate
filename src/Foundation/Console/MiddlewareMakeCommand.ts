/**
 * (c) Phan Trung NguyÃªn <nguyenpl117@gmail.com>
 * User: nguyenpl117
 * Date: 3/14/2020
 * Time: 4:26 PM
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 */
import { Application } from '../Application';
import * as path from 'path';
import { GeneratorCommand } from './GeneratorCommand';
import { args } from '@tngraphql/console';

export class MiddlewareMakeCommand extends GeneratorCommand {
    protected getSuffix(): string {
        return 'middleware';
    }
    /**
     * Command name. The command will be registered using this name only. Make
     * sure their aren't any spaces inside the command name.
     */
    static commandName: string = 'make:middleware';

    /**
     * The description of the command displayed on the help screen.
     * A good command will always have some description.
     */
    static description: string = 'Create a new middleware class';

    @args.string()
    public name: string;

    protected getStub() {
        return path.join(__dirname, 'stub/middleware.stub');
    }

    async handle(...args: any[]): Promise<boolean> {
        return super.handle(...args);
    }

    getDestinationPath() {
        return path.join(this.application.getBasePath(), 'app/GraphQL/Middleware');
    }
}
