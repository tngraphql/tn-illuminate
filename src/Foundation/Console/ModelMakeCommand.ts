/**
 * (c) Phan Trung NguyÃªn <nguyenpl117@gmail.com>
 * User: nguyenpl117
 * Date: 3/14/2020
 * Time: 4:26 PM
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 */
import { GeneratorCommand } from './GeneratorCommand';
import * as path from 'path';
import { args } from 'tn-console';

export class ModelMakeCommand extends GeneratorCommand {
    protected getStub(): string {
        return path.join(__dirname, 'stub/model.stub');
    }

    protected getSuffix(): string {
        return 'model';
    }
    /**
     * Command name. The command will be registered using this name only. Make
     * sure their aren't any spaces inside the command name.
     */
    static commandName: string = 'make:model';

    static description: string = 'Make a new Lucid model';

    @args.string()
    public name: string;

    getDestinationPath() {
        return path.join(this.application.getBasePath(), 'app');
    }
}
