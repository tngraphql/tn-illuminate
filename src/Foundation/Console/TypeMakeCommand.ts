/**
 * (c) Phan Trung NguyÃªn <nguyenpl117@gmail.com>
 * User: nguyenpl117
 * Date: 3/14/2020
 * Time: 4:26 PM
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 */
import { GeneratorCommand } from './GeneratorCommand';
import * as path from 'path';
import { args } from 'tn-console';

export class TypeMakeCommand extends GeneratorCommand  {
    protected getSuffix(): string {
        return 'type';
    }

    protected getStub(): string {
        return path.join(__dirname, 'stub/type.stub');
    }
    /**
     * Command name. The command will be registered using this name only. Make
     * sure their aren't any spaces inside the command name.
     */
    static commandName: string = 'make:type';

    static description: string = 'Make a new graphql objectType';

    @args.string()
    public name: string;

    getDestinationPath() {
        return path.join(this.application.getBasePath(), 'app/GraphQL/Types');
    }

    async handle(...args: any[]): Promise<boolean> {
        return super.handle(...args);
    }

}