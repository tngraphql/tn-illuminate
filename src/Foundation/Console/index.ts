/**
 * (c) Phan Trung NguyÃªn <nguyenpl117@gmail.com>
 * User: nguyenpl117
 * Date: 3/14/2020
 * Time: 10:26 AM
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 */

export * from './ConsoleKernel';
export * from './AceApplication';
export * from './CommandMakeCommand';
export * from './GeneratorCommand';
export * from './KeyGenerateCommand';
export * from './MiddlewareMakeCommand';
export * from './MigrateMakeCommand';
export * from './ModelMakeCommand';
export * from './ProviderMakeCommand';
export * from './ResolveMakeCommand';
export * from './TypeMakeCommand';
