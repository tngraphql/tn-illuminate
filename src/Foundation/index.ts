/**
 * (c) Phan Trung NguyÃªn <nguyenpl117@gmail.com>
 * User: nguyenpl117
 * Date: 3/11/2020
 * Time: 12:35 PM
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 */

export { BootProviders, LoadConfiguration, LoadEnvironmentVariables, RegisterProviders } from './Bootstrap';
export { Kernel } from './Kernel';
export { Application } from './Application';
export * from './Console';
export * from './GraphQL';