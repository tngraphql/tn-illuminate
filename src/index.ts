/**
 * (c) Phan Trung NguyÃªn <nguyenpl117@gmail.com>
 * User: nguyenpl117
 * Date: 3/11/2020
 * Time: 12:31 PM
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 */

export { Repository } from './config/Repository';
export { Container, Injector, InvalidArgumentException, BindingResolutionException, LogicException } from './Container';
export *  from './Decorators';
export {
    Application, ConsoleKernel, BootProviders, RegisterProviders, LoadEnvironmentVariables, LoadConfiguration
} from './Foundation';
export { ServiceProvider } from './Support/ServiceProvider';
export { ensureIsFunction, isClass, isFuntion, isEsm, isObject } from './utils';
export * from '@tngraphql/console';
