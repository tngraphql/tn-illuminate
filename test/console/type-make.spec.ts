/**
 * (c) Phan Trung NguyÃªn <nguyenpl117@gmail.com>
 * User: nguyenpl117
 * Date: 3/14/2020
 * Time: 10:33 PM
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 */

import { AceApplication, ConsoleKernel } from '../../src/Foundation/Console';
import { Application } from '../../src/Foundation';
import { Filesystem } from '@poppinss/dev-utils/build';
import { join } from "path";
import { MiddlewareMakeCommand } from '../../src/Foundation/Console/MiddlewareMakeCommand';
import { ModelMakeCommand } from '../../src/Foundation/Console/ModelMakeCommand';
import { TypeMakeCommand } from '../../src/Foundation/Console/TypeMakeCommand';

const fs = new Filesystem(join(__dirname, './type'))
describe('Type Make', () => {

    afterEach(async () => {
        jest.resetModules();
        await fs.cleanup();
    });

    beforeAll(async () => {
        AceApplication.starting(ace => {
            ace.register([TypeMakeCommand]);
        });
    });

    beforeEach(async () => {
        await fs.add('config/app.ts', `export default {
            providers: []
        }`);
    })

    it('make a new type', async () => {
        const app = new Application(fs.basePath);

        const kernel: ConsoleKernel = await app.make<ConsoleKernel>(ConsoleKernel);

        await kernel.call('make:type', ['user']);

        expect(await fs.get('app/GraphQL/Types/User.ts')).toMatchSnapshot()
    });
});