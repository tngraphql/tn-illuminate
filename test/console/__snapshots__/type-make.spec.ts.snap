// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Type Make make a new type 1`] = `
"import { Field, ObjectType } from 'tn-graphql';

@ObjectType()
export class User {
    @Field()
    id: number

    @Field()
    name: string
}"
`;
